lib/pair.ts

```diff
@@ -0,0 +1,266 @@
+import * as bitcoin from 'bitcoinjs-lib';
+import {encodeBtcAddress} from './encode';
+import {BigNumberish, BytesLike, BigNumber} from 'ethers';
+import {Obligation} from '../typechain/Obligation';
+import {Option} from '../typechain/Option';
+import {ITreasury} from '../typechain/ITreasury';
+import {OptionLib} from '../typechain/OptionLib';
+import {Ierc20} from '../typechain/Ierc20';
+import {BitcoinPriceFeed} from './feed';
+import {BtcAddress} from './bitcoin';
+
+interface EuropeanOption {
```

Comment:
Do we want to make the fields `readonly`?

Reply:
We could except for the "spotPrice"